src/routes/accounts.py

```
@@ -1,25 +1,243 @@
+async def login(
+    model: LoginSchema,
+    db: AsyncSession = Depends(get_db),
+    jwt_create: JWTAuthManager = Depends(get_jwt_manager),
+):
+    try:
+        user_auth = await db.execute(
+            select(UserModel).where(UserModel.email == model.email)
+        )
+        user = user_auth.scalar_one_or_none()
+        if user is None:
+            raise HTTPException(
+                status_code=401, detail="Invalid email or password."
+            )
+        password_verification = verify_password(
+            model.password, user._hashed_password
+        )
+        if not password_verification:
+            raise HTTPException(
+                status_code=401, detail="Invalid email or password."
+            )
+        if not user.is_active:
+            raise HTTPException(
+                status_code=403, detail="User account is not activated."
+            )
+
+        token_data = {"user_id": user.id}
+        access = jwt_create.create_access_token(token_data)
+        refresh = jwt_create.create_refresh_token(token_data)
+        new_refresh_token = RefreshTokenModel(token=refresh, user_id=user.id)
+        db.add(new_refresh_token)
+        await db.commit()
+        await db.refresh(new_refresh_token)
+
+        return LoginResponseSchema(
+            access_token=access, refresh_token=refresh, token_type="bearer"
+        )
+    except HTTPException:
+        raise
+    except Exception:
+        raise HTTPException(
+            status_code=500,
+            detail="An error occurred while processing the request.",
+        )
+
+
+@router.post("/refresh/", response_model=dict)
+async def get_new_token(
+    model: TokenRefresh,
+    db: AsyncSession = Depends(get_db),
+    jwt_create: JWTAuthManager = Depends(get_jwt_manager),
+):
+    try:
+        decode_token = jwt_create.decode_refresh_token(model.refresh_token)
+    except Exception as e:
+        raise HTTPException(status_code=400, detail=str(e))
```

In the refresh endpoint you catch all exceptions when decoding the token and return `str(e)` as the error detail. The token manager raises specific security exceptions (e.g., TokenExpiredError, InvalidTokenError). Catch those explicitly and map them to the exact messages the tests expect (for example, on TokenExpiredError return 400 with detail "Token has expired."). Avoid returning generic `str(e)` which may be empty or non-deterministic.